From 7aed0014a1d81805fc5f16f80f236660f4ad18c1 Mon Sep 17 00:00:00 2001
From: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Date: Tue, 1 Dec 2015 15:50:12 +0530
Subject: [PATCH 0305/1566] media: xilinx: Use xvip_write to update background
 pattern

Use xvip_write to update background pattern.
It allows to have pattern count > mask.

Signed-off-by: Radhey Shyam Pandey <radheys@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Tested-by: Christian Kohn <christian.kohn@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/media/platform/xilinx/xilinx-tpg.c |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-tpg.c b/drivers/media/platform/xilinx/xilinx-tpg.c
index a25120d..763a23f 100644
--- a/drivers/media/platform/xilinx/xilinx-tpg.c
+++ b/drivers/media/platform/xilinx/xilinx-tpg.c
@@ -232,7 +232,6 @@ static int xtpg_s_stream(struct v4l2_subdev *subdev, int enable)
 	unsigned int height = xtpg->formats[0].height;
 	bool passthrough;
 	u32 bayer_phase;
-	u32 xtpg_pattern_offset;
 
 	if (!enable) {
 		if (xtpg->is_hls)
@@ -266,10 +265,8 @@ static int xtpg_s_stream(struct v4l2_subdev *subdev, int enable)
 		xvip_write(&xtpg->xvip, XTPG_HLS_COLOR_FORMAT, fmt);
 		xvip_write(&xtpg->xvip, XHLS_REG_COLS, width);
 		xvip_write(&xtpg->xvip, XHLS_REG_ROWS, height);
-		xtpg_pattern_offset = XTPG_HLS_BG_PATTERN;
 	} else {
 		xvip_set_frame_size(&xtpg->xvip, &xtpg->formats[0]);
-		xtpg_pattern_offset = XTPG_PATTERN_CONTROL;
 	}
 
 	if (xtpg->vtc)
@@ -282,7 +279,11 @@ static int xtpg_s_stream(struct v4l2_subdev *subdev, int enable)
 	 */
 	mutex_lock(xtpg->ctrl_handler.lock);
 
-	xvip_clr_and_set(&xtpg->xvip, xtpg_pattern_offset,
+	if (xtpg->is_hls)
+		xvip_write(&xtpg->xvip, XTPG_HLS_BG_PATTERN,
+			   xtpg->pattern->cur.val);
+	else
+		xvip_clr_and_set(&xtpg->xvip, XTPG_PATTERN_CONTROL,
 			 XTPG_PATTERN_MASK, xtpg->pattern->cur.val);
 
 	/*
-- 
1.7.5.4

